File: models/train_model.R

library(randomForest)

train_model <- function(train_data){

  model <- randomForest(
    Churn ~ .,
    data=train_data,
    ntree=200,
    importance=TRUE
  )

  saveRDS(model, "models/churn_model.rds")

  return(model)
}

File: models/train_model.py
import joblib
from sklearn.ensemble import RandomForestClassifier
import pandas as pd


def train_model(train_data):
    X = train_data.drop(columns=['Churn'])
    y = train_data['Churn']

    model = RandomForestClassifier(n_estimators=200)
    model.fit(X, y)

    joblib.dump(model, 'models/churn_model.pkl')

    return model

if __name__ == "__main__":
    # Example usage
    train_data = pd.read_csv('path_to_train_data.csv')
    train_model(train_data)
